guillotina/db/walking.py

```diff
@@ -0,0 +1,49 @@
+from guillotina.configure import adapter
+from guillotina.interfaces import IFolder
+from guillotina.interfaces import IIteratorResources
+from guillotina.interfaces import IResource
+
+
+@adapter(for_=(IFolder), provides=IIteratorResources)
+class WalkingIterator(object):
+    def __init__(self, context):
+        self.context = context
+        self.total = 0
+        self.count = 0
+
+    async def deep_walk(self, actual):
+        self.total += await actual.async_len()
```

Comment:
I would recommend against using `async_len`. PG is always very slow to get total counts.